clang/lib/Sema/SemaDecl.cpp

```diff
@@ -8732,6 +8732,18 @@ void Sema::CheckVariableDeclarationType(VarDecl *NewVD) {
     }
   }
 
+  // zero sized static arrays are not allowed in HIP device functions
+  if (getLangOpts().HIP &&
+      DeclAttrsMatchCUDAMode(getLangOpts(), getCurFunctionDecl())) {
```

Comment:
Do you need to check LangOpts.CUDAIsDevice?

Reply:
Yes and that is being checked in the function DeclAttrsMatchCUDAMode.

Reply:
The check there is `isDeviceSideDecl == LangOpts.CUDAIsDevice`... but you don't want to diagnose this on host, do you?  (The regression test only checks the -fcuda-is-device case.)

Reply:
Yes, that's right, I dont want to diagnose this on host, but the device side decl are cuda device specific. So I thought there would not be any problem.

Reply:
else it would be cleaner to go back to how I was checking previously as this is the only helper function I found that is closely related.

Reply:
The code, in its current state, *will* diagnose on host.  (The testcase doesn't show this because it only runs with `-fcuda-is-device`.)